provision/acc_provision/acc_provision.py

```diff
@@ -1628,9 +1653,26 @@ def generate_kube_yaml(args, config, operator_output, operator_tar, operator_cr_
 
 
 def generate_apic_config(flavor_opts, config, prov_apic, apic_file):
+    # Used (_) underscore in version to support getattr api that considers (.) dot as a different folder
+    if config.get("target_version"):
+        if (parse_version.parse(config["target_version"]) <= parse_version.parse('5.2.3.3')):
+            ver = "5_2_3_3"
+        elif (parse_version.parse(config["target_version"]) == parse_version.parse('5.2.3.4')):
+            ver = "5_2_3_4"
+        elif parse_version.parse(config["target_version"]) == parse_version.parse('5.2.3.5'):
+            ver = "5_2_3_5"
+        elif parse_version.parse(config["target_version"]) >= parse_version.parse('5.2.3.6'):
+            ver = "5_2_3_6"
+    else:
```

Comment:
I think I had asked earlier, why does this default to 5.2.3.3? Should it not default to the latest, and preferably, should this not be coming from a "default" config which we can change as desired as opposed to revisiting this code? (sorry if I am missing something basic)

Reply:
this was default to 5.2.3.3 if no target_version is defined in input_file. If we want to default to the latest version then sure I can do that. I was having the impression to default it to the last stable version

Reply:
Lets also pick up this value from the place where you are picking up the other default values. Also, can we not generically parse this version and construct the version name with the underscore in it? That way we just need to change the default value in the config whenever the new CNI version is released, and not code change is required here. If the user enters a version which is not available in the templates, the template lookup should error out with the appropriate message that the version is not supported.

Reply:
there is a default_version in the config, I missed that the target_version is not passed in the config today. default_version comes from the flavor and is always set. You should use this instead of target_version.

Reply:
ok that make sense about the default version. Let me check that as well